Simplenote/MetricsViewController.swift

```diff
@@ -138,21 +143,186 @@ private extension MetricsViewController {
 extension MetricsViewController: EntityObserverDelegate {
 
     func entityObserver(_ observer: EntityObserver, didObserveChanges for: Set<NSManagedObjectID>) {
-        refreshMetrics()
+        refreshInterface()
     }
 }
 
 
-// MARK: - Rendering Metrics!
+// MARK: - Refreshing!
 //
 private extension MetricsViewController {
 
-    func refreshMetrics() {
+    func refreshInterface() {
+        rows = metricRows(for: notes) + referenceRows(from: resultsController.fetchedObjects)
+        adjustRootViewSize()
+        tableView.reloadData()
+    }
+
+    func adjustRootViewSize() {
+        view.frame.size = calculatePreferredSize(for: rows)
+    }
+
+    /// Returns Metric Rows for a collection of notes
+    ///
+    func metricRows(for notes: [Note]) -> [Row] {
         let metrics = NoteMetrics(notes: notes)
+        return [
+            .header(text: NSLocalizedString("Information", comment: "Note Metrics Title")),
+            .metric(title: NSLocalizedString("Modified", comment: "Note Modification Date"),
+                    value: metrics.modifiedDate),
 
-        modifiedDetailsLabel.stringValue = metrics.modifiedDate ?? "-"
-        createdDetailsLabel.stringValue = metrics.creationDate ?? "-"
-        wordsDetailsLabel.stringValue = String(metrics.numberOfWords)
-        charsDetailsLabel.stringValue = String(metrics.numberOfChars)
+            .metric(title: NSLocalizedString("Created", comment: "Note Creation Date"),
+                    value: metrics.creationDate),
+
+            .metric(title: NSLocalizedString("Words", comment: "Number of words in the note"),
+                    value: String(metrics.numberOfWords)),
+
+            .metric(title: NSLocalizedString("Characters", comment: "Number of characters in the note"),
+                    value: String(metrics.numberOfChars))
```

Comment:
Maybe we can format numbers using a decimal formatter. If the number is big, it looks better and easier to read with delimiters.

Reply:
Awesome idea, done!! thank you!